tests/model_registry/conftest.py

```diff
@@ -303,3 +319,163 @@ def generated_schema(model_registry_instance_rest_endpoint: str) -> Any:
         uri="https://raw.githubusercontent.com/kubeflow/model-registry/main/api/openapi/model-registry.yaml",
         base_url=f"https://{model_registry_instance_rest_endpoint}/",
     )
+
+
+def pytest_sessionstart(session: pytest.Session) -> None:
+    """Downloads external tests into the local project structure before test execution."""
+    if session.config.option.model_registry_upstream:
+        LOGGER.info("Model Registry upstream tests are being fetched due to --model-registry-upstream flag")
+        fetch_external_tests_from_github(
+            dir=UPSTREAM_TESTS_DIR, files=FILES, repo_api_url=REPO_API_URL, branch=REPO_BRANCH, file_path=FILE_PATH
+        )
+    else:
+        LOGGER.info("Skipping upstream test fetching for Model registry. Enable with --model-registry-upstream")
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_sessionfinish(session: pytest.Session, exitstatus: pytest.ExitCode) -> None:
+    """Clean up upstream tests from kubeflow/model-registry"""
+    if session.config.option.model_registry_upstream:
+        if os.path.exists(UPSTREAM_TESTS_DIR):
+            shutil.rmtree(UPSTREAM_TESTS_DIR)
+            LOGGER.info(f"Removed upstream model registry tests from {UPSTREAM_TESTS_DIR}")
+
+
+# fixture needed for upstream tests
+@pytest.fixture(scope="function")
+def client(model_registry_instance_rest_endpoint: str, current_client_token: str) -> ModelRegistryClient:
```

Comment:
You might want to rename it to model_registry_client or something more meaningful. It is too broad

Reply:
Thank you for the explaination.